src/cmap/auth/mongodb_oidc.ts

```diff
@@ -71,10 +72,19 @@ export interface Workflow {
   execute(
     connection: Connection,
     credentials: MongoCredentials,
-    reauthenticating: boolean,
+    cache?: TokenCache,
     response?: Document
   ): Promise<Document>;
 
+  /**
+   * Each workflow should specify the correct custom behaviour for reauthentication.
+   */
+  reauthenticate(
+    connection: Connection,
```

Comment:
No.

Authentication and reauthentication both happen after the initial handshake (with the exception of speculative auth, but we don't send any commands through Connection.command() when preparing a handshake document for speculative authentication.

auto encryption is irrelevant to authentication because user data isn't being stored (what data would FLE be encrypting for a saslStart or saslContinue?) and [authentication commands are bypass encryption](https://github.com/mongodb/libmongocrypt/blob/ba2443f1c940f46de41fededf9217ca1da0a19a9/src/mongocrypt-ctx-encrypt.c#L2780).

Reply:
Oh, excellent thank you for digging up that line of code.